src/app/loadout-drawer/LoadoutDrawer2.tsx

```diff
@@ -75,6 +96,42 @@ export default function LoadoutDrawer2() {
     }, [])
   );
 
+  const currentStoreId = getCurrentStore(stores)?.id;
+  const storeIdsByClassType = useSelector(storeIdsByClassTypeSelector);
+
+  // Load in a full loadout specified in the URL
+  useEffect(() => {
+    if (!stores.length || !defs?.isDestiny2()) {
+      return;
+    }
+    const searchParams = new URLSearchParams(queryString);
+    const loadoutJSON = searchParams.get('loadout');
+    if (loadoutJSON) {
+      const parsedLoadout = convertDimApiLoadoutToLoadout(JSON.parse(loadoutJSON));
+      if (parsedLoadout) {
+        const storeId =
+          parsedLoadout.classType === DestinyClass.Unknown
+            ? currentStoreId
+            : storeIdsByClassType[parsedLoadout.classType];
+
+        if (!storeId) {
+          warnMissingClass(parsedLoadout.classType, defs);
+          return;
+        }
+
+        parsedLoadout.id = uuidv4();
+
+        stateDispatch({
+          type: 'editLoadout',
+          loadout: parsedLoadout,
+          storeId,
+          showClass: false,
+          isNew: true,
+        });
+      }
+    }
+  }, [defs, queryString, currentStoreId, storeIdsByClassType, stores.length]);
```

Comment:
I'd suggest removing the loadout from the URL once opened. If we don't remove the loadout from the URL, we're going to have to be very careful with this dependency array because any changes will cause the shared loadout being edited to reset to the shared loadout (or reopen the shared loadout while the user is doing something else on this page).

Reply:
That's a good suggestion.